runtime/balrt.h

```diff
@@ -368,10 +440,54 @@ static inline TaggedPtr ptrAddShiftedTag(UntypedPtr tp, uint64_t shiftedTag) {
     return (TaggedPtr)p;
 }
 
+static READONLY inline int64_t arrayCompare(TaggedPtr lhs, TaggedPtr rhs, int64_t(*comparator)(TaggedPtr, TaggedPtr)) {
+    bool isLhsNull = lhs == NULL;
+    bool isRhsNull = rhs == NULL;
+    if (isLhsNull && isRhsNull) {
+        return EQ;
+    }
+    else if (isLhsNull || isRhsNull) {
+        return UN;
+    }
+    ListPtr lhsListPtr = taggedToPtr(lhs);
+    ListPtr rhsListPtr = taggedToPtr(rhs);
+    int64_t lhsLen = lhsListPtr->tpArray.length;
+    int64_t rhsLen = rhsListPtr->tpArray.length;
+    int64_t length;
+    if (lhsLen <= rhsLen) {
+        length = lhsLen;
+    }
+    else {
+        length = rhsLen;
+    }
+    GC TaggedPtr* lhsArr = lhsListPtr->tpArray.members;
```

Comment:
`TaggedPtr *lhsArr`

Reply:
If we drop the GC part it throws an error saying it changes the address space of a pointer
```
leads to error `initializing 'TaggedPtr *' (aka '__attribute__((address_space(1))) char **') with an expression of type 'GC TaggedPtr *__attribute__((address_space(1)))' (aka '__attribute__((address_space(1))) char *__attribute__((address_space(1))) *__attribute__((address_space(1)))') changes address space of pointer`

```

Reply:
I mean space before not after the `*`.

Reply:
fixed